File: Arcade/Intro/Java/BuildPalindrome.java
final class BuildPalindrome {

	String reverseString(String str) {
	    return new StringBuilder(str).reverse().toString();
	}

	String buildPalindrome(String st) {
	    if(reverseString(st).equals(st)) return st;
	    
	    String sg = "", out = "";
	    for(char c : st.toCharArray()) {
	        out = st+reverseString(sg += c+"");
	        if(out.equals(reverseString(out))) break;
	    }
	    return out;
	}
}

File: Arcade/Intro/javascript/buildPalindrome.js
function reverse(s) {
    let out = "";
    for(let i = s.length - 1; i >= 0; i--) out += s[i];
    return out;
}

function buildPalindrome(st) {
    if(reverse(st) === st) return st;
    
    let out = "", sg = "";
    for(let ch of st) {
        out = st+reverse(sg += ch);
        if(reverse(out) === out) break;
    }
    
    return out;
}